pages/graphviz.php

```diff
@@ -142,15 +142,15 @@ function GraphvizLifecycle($sClass)
 	}
 	else
 	{
-		header('Content-type: image/png');
-		echo file_get_contents($sImageFilePath);
+		header('Content-type: image/svg+xml');
+		readfile($sImageFilePath);
 	}
 	@unlink($sDotFilePath);
```

Comment:
PS, why is the dot file being removed and the generated image isn't?

Shouldn't both be removed? Or keep them both?

Reply:
Hello Thomas,

Out of the box there are already some pre-generated lifecycle images for the existing classes (UserRequest / Incident). They are used as a fallback if you don't have GraphViz installed and a standard DM. That's why the file isn't deleted I think, otherwise it would only work the first time.

That being said, it would be clearer if the fallback image was in a dedicated variable. Also if the image isn't deleted afterwards, we should check before generating it again and reuse it.

Reply:
I have seen the fallback image, but that one is on a different location (in the module) as well. I'm clearly only talking about the generated image only.

Reply:
@Molkobain ?

Reply:
Hello @Hipska , are you replying about the thread or the status or the PR?

Reply:
This actual thread. I'm still looking for an answer to my initial question from 8 days ago.

Reply:
I think it was for some caching purpose but it's clearly not using it well as the image is re-generated every time. As Denis is on vacation I can't ask him what the intention was.

Reply:
Can you try again now?

Reply:
@dflaven any input on this?

Reply:
Can you contact @dflaven directly internally?

Reply:
Any response from @dflaven ?

Reply:
Hello @Hipska , just talked with him IRL. It seems to be a mistake from the past, you can fix it 😁

Reply:
Keep both? Or remove both?

Reply:
Remove both

Reply:
This seems not to be done yet, could you quickly do it in a commit? Or do I need to create a new branch and PR for that?

```suggestion
	@unlink($sDotFilePath);
	@unlink($sImageFilePath);
```

Reply:
I thought you would make a PR, but sure just did it in 7613ac7 😊